tests/Lykke.AlgoStore.Tests/Unit/CodeBuildTests.cs

```diff
@@ -0,0 +1,182 @@
9uDQp7DQogICAgLy8vIDxzdW1tYXJ5Pg0KICAgIC8vLyBSRU1BUks6IEp1c3QgYSBkdW1teSBhbGdvIGltcGxlbWVudGF0aW9uIGZvciBmdXR1cmUgcmVmZXJlbmNlLg0KICAgIC8vLyBXZSBjYW4gYW5kIHdpbGwgcmVtb3ZlIHRoaXMgd2hlbiBmaXJzdCBhbGdvIGlzIGltcGxlbWVudGVkIDopDQogICAgLy8vIDwvc3VtbWFyeT4NCiAgICBzZWFsZWQgY2xhc3MgRHVtbXlBbGdvIDogQmFzZUFsZ28NCiAgICB7DQogICAgICAgIHB1YmxpYyBTbWFGdW5jdGlvbiBfc2hvcnRTbWE7DQogICAgICAgIHB1YmxpYyBTbWFGdW5jdGlvbiBfbG9uZ1NtYTsNCg0KICAgICAgICBwdWJsaWMgb3ZlcnJpZGUgdm9pZCBPblN0YXJ0VXAoSUZ1bmN0aW9uUHJvdmlkZXIgZnVuY3Rpb25zKQ0KICAgICAgICB7DQoNCiAgICAgICAgICAgIF9zaG9ydFNtYSA9IGZ1bmN0aW9ucy5HZXRGdW5jdGlvbjxTbWFGdW5jdGlvbj4oIlNNQV9TaG9ydCIpOw0KICAgICAgICAgICAgX2xvbmdTbWEgPSBmdW5jdGlvbnMuR2V0RnVuY3Rpb248U21hRnVuY3Rpb24+KCJTTUFfTG9uZyIpOw0KICAgICAgICB9DQoNCiAgICAgICAgcHVibGljIG92ZXJyaWRlIHZvaWQgT25RdW90ZVJlY2VpdmVkKElRdW90ZUNvbnRleHQgY29udGV4dCkNCiAgICAgICAgew0KICAgICAgICAgICAgY29udGV4dC5BY3Rpb25zLkxvZygkIlZvbHVtZSB2YWx1ZToge1ZvbHVtZX0iKTsNCg0KICAgICAgICAgICAgdmFyIHF1b3RlID0gY29udGV4dC5EYXRhLlF1b3RlOw0KICAgICAgICAgICAgY29udGV4dC5BY3Rpb25zLkxvZygkIlJlY2VpdmluZyBxdW90ZSBhdCB7RGF0ZVRpbWUuVXRjTm93fSAiICsNCiAgICAgICAgICAgICAgICAkInt7cXVvdGUuUHJpY2U6IHtxdW90ZS5QcmljZX19fSwge3txdW90ZS5UaW1lc3RhbXA6IHtxdW90ZS5UaW1lc3RhbXB9fX0sICIgKw0KICAgICAgICAgICAgICAgICQie3txdW90ZS5Jc0J1eToge3F1b3RlLklzQnV5fX19LCB7e3F1b3RlLklzT25saW5lOiB7cXVvdGUuSXNPbmxpbmV9fX0iKTsNCg0KICAgICAgICAgICAgdmFyIHNtYVNob3J0ID0gX3Nob3J0U21hLlZhbHVlOw0KICAgICAgICAgICAgdmFyIHNtYUxvbmcgPSBfbG9uZ1NtYS5WYWx1ZTsNCiAgICAgICAgICAgIGNvbnRleHQuQWN0aW9ucy5Mb2coJCJGdW5jdGlvbiB2YWx1ZXMgYXJlOiBTTUFfU2hvcnQ6IHtzbWFTaG9ydH0sIFNNQV9Mb25nOiB7c21hTG9uZ30iKTsNCg0KICAgICAgICAgICAgLy9pZiAocXVvdGUuUHJpY2UgPCAxMDAwMCkNCiAgICAgICAgICAgIC8vew0KICAgICAgICAgICAgY29udGV4dC5BY3Rpb25zLkJ1eShWb2x1bWUpOw0KICAgICAgICAgICAgLy99DQoNCiAgICAgICAgICAgIC8vaWYgKHF1b3RlLlByaWNlID4gNzAwMCkNCiAgICAgICAgICAgIC8vew0KICAgICAgICAgICAgLy8gICAgY29udGV4dC5BY3Rpb25zLlNlbGwoVm9sdW1lKTsNCiAgICAgICAgICAgIC8vfQ0KICAgICAgICB9DQoNCg0KICAgICAgICBwdWJsaWMgb3ZlcnJpZGUgdm9pZCBPbkNhbmRsZVJlY2VpdmVkKElDYW5kbGVDb250ZXh0IGNvbnRleHQpDQogICAgICAgIHsNCiAgICAgICAgICAgIGNvbnRleHQuQWN0aW9ucy5Mb2coJCJWb2x1bWUgdmFsdWU6IHtWb2x1bWV9Iik7DQoNCiAgICAgICAgICAgIHZhciBjYW5kbGUgPSBjb250ZXh0LkRhdGEuQ2FuZGxlOw0KICAgICAgICAgICAgY29udGV4dC5BY3Rpb25zLkxvZygkIlJlY2VpdmluZyBjYW5kbGUgYXQge2NhbmRsZS5EYXRlVGltZX0gY2FuZGxlIGNsb3NlIFByaWNlIHtjYW5kbGUuQ2xvc2V9Iik7DQoNCiAgICAgICAgICAgIHZhciBzbWFTaG9ydCA9IF9zaG9ydFNtYS5WYWx1ZTsNCiAgICAgICAgICAgIHZhciBzbWFMb25nID0gX2xvbmdTbWEuVmFsdWU7DQogICAgICAgICAgICBjb250ZXh0LkFjdGlvbnMuTG9nKCQiRnVuY3Rpb24gdmFsdWVzIGFyZTogU01BX1Nob3J0OiB7c21hU2hvcnR9LCBTTUFfTG9uZzoge3NtYUxvbmd9Iik7DQoNCiAgICAgICAgICAgIC8vaWYgKHF1b3RlLlByaWNlIDwgMTAwMDApDQogICAgICAgICAgICAvL3sNCiAgICAgICAgICAgIC8vY29udGV4dC5BY3Rpb25zLkJ1eShjb250ZXh0LkRhdGEuQ2FuZGxlLCBWb2x1bWUpOw0KICAgICAgICAgICAgLy99DQoNCiAgICAgICAgICAgIC8vaWYgKHF1b3RlLlByaWNlID4gNzAwMCkNCiAgICAgICAgICAgIC8vew0KICAgICAgICAgICAgY29udGV4dC5BY3Rpb25zLlNlbGwoY29udGV4dC5EYXRhLkNhbmRsZSwgVm9sdW1lKTsNCiAgICAgICAgICAgIC8vfQ0KICAgICAgICB9DQogICAgfQ0KfQ==";
```

Comment:
Consider writing the minimal implementation needed (in plaintext) to test the code instead of pasting a massive base64 chunk